plugins/core/profile.go

```diff
@@ -0,0 +1,341 @@
+	// Start a goroutine to supplement each data chunk with business information
+	go func() {
+		for rawResult := range rawCh {
+			d = append(d, rawResult.data...)
+			m.mu.Lock()
+			// Get business information from currentTask
+			task := m.currentTask
+			m.mu.Unlock()
+
+			if task == nil {
+				m.Log.Info("no task")
+				continue // Task has ended, ignore
+			}
+
+			if rawResult.isLast {
+				m.FinalReportResults <- reporter.ProfileResult{
+					TaskID:  task.taskID,
+					Payload: rawResult.data,
+					IsLast:  rawResult.isLast,
+				}
+				m.mu.Lock()
+				m.TraceProfileTasks[m.currentTask.taskID].Status = reporter.Finished
+				m.currentTask = nil
+				m.profileEvents.BaseEventStatus[CurTaskExist] = false
+				m.mu.Unlock()
+			} else {
+				m.FinalReportResults <- reporter.ProfileResult{
+					TaskID:  task.taskID,
+					Payload: rawResult.data,
+					IsLast:  rawResult.isLast,
+				}
+			}
+		}
+	}()
+}
+
+func NewProfileManager(log operator.LogOperator) *ProfileManager {
+	pm := &ProfileManager{
+		TraceProfileTasks:  make(map[string]*reporter.TraceProfileTask),
+		FinalReportResults: make(chan reporter.ProfileResult, maxSendQueueSize),
+		profileEvents:      NewEventManager(),
+	}
+	pm.RegisterProfileEvents()
+
+	if log == nil {
+		log = newDefaultLogger()
+	}
+	pm.Log = log
+	pm.initReportChannel()
+	pm.profilingWriter = NewProfilingWriter(
+		ChunkSize,
+		pm.rawCh,
+	)
+	return pm
+}
+
+func (m *ProfileManager) AddProfileTask(args []*common.KeyStringValuePair) {
+	m.mu.Lock()
+	defer m.mu.Unlock()
+	var task reporter.TraceProfileTask
+	for _, arg := range args {
+		switch arg.Key {
+		case "TaskID":
+			task.TaskID = arg.Value
+		case "EndpointName":
+			task.EndpointName = arg.Value
+		case "Duration":
+			// Duration min
+			task.Duration = parseInt(arg.Value)
+		case "MinDurationThreshold":
+			task.MinDurationThreshold = parseInt64(arg.Value)
+		case "DumpPeriod":
+			task.DumpPeriod = parseInt(arg.Value)
+		case "MaxSamplingCount":
+			task.MaxSamplingCount = parseInt(arg.Value)
+		case "StartTime":
+			task.StartTime = parseInt64(arg.Value)
+		case "CreateTime":
+			task.CreateTime = parseInt64(arg.Value)
+		case "SerialNumber":
+			task.SerialNumber = arg.Value
+		}
+	}
+	m.Log.Info("adding profile task:", task)
+	if _, exists := m.TraceProfileTasks[task.TaskID]; exists {
+		return
+	}
+	endTime := task.StartTime + int64(task.Duration)*60*1000
+	task.EndTime = endTime
+	task.Status = reporter.Pending
+	m.TraceProfileTasks[task.TaskID] = &task
+	m.TrySetCurrentTask(&task)
+	m.tryStartCPUProfiling()
+}
+
+func (m *ProfileManager) RemoveProfileTask() {
+	m.mu.Lock()
+	defer m.mu.Unlock()
+	for k, t := range m.TraceProfileTasks {
+		if t.Status == reporter.Reported || t.EndTime < time.Now().Unix() {
+			delete(m.TraceProfileTasks, k)
+		}
+	}
+}
+
+func (m *ProfileManager) tryStartCPUProfiling() {
+	ok, err := m.profileEvents.ExecuteComplexEvent(CouldProfile)
+	if err != nil {
+		m.Log.Errorf("profile event error:%v", err)
```

Comment:
What will happen when the `TrySetCurrentTask` executes failure? Will the start profiling keep running? You can consolidate them all into a single method to simplify the logic of task execution.